src/main/java/org/opensearch/sdk/handlers/EnvironmentSettingsResponseHandler.java

```diff
@@ -11,27 +11,42 @@
 import org.apache.logging.log4j.Logger;
 import org.opensearch.env.EnvironmentSettingsResponse;
 import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.settings.Settings;
 import org.opensearch.sdk.ExtensionsRunner;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.TransportException;
 import org.opensearch.transport.TransportResponseHandler;
 
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
 
 /**
  * This class handles the response from OpenSearch to a {@link ExtensionsRunner#sendEnvironmentSettingsRequest} call.
  */
 public class EnvironmentSettingsResponseHandler implements TransportResponseHandler<EnvironmentSettingsResponse> {
     private static final Logger logger = LogManager.getLogger(EnvironmentSettingsResponseHandler.class);
 
+    private final CountDownLatch inProgressLatch;
+    private Settings environmentSettings;
+
+    public EnvironmentSettingsResponseHandler(CountDownLatch inProgressLatch) {
+        this.inProgressLatch = inProgressLatch;
```

Comment:
We should not pass latch as a part of constructor. It should be use where we are initializing the object for this class

Reply:
Sure ill make the modification

Reply:
@owaiskazi19 do you mean pass it in the request?

Reply:
I meant outside this class where we are creating the object of this class or in the class itself to make it independent